JavaScript/call-apply-bind.md

```diff
@@ -0,0 +1,148 @@
+</br>
+</br>
+
+
+Using the ```bind``` method we are able to make a copy of any function that we call ```bind``` on, which has its ```this``` variable pointing to an object that we specify.
+
+The code below copies the ```printUser``` function, with the copy configured to have its ```this``` variable pointed at the ```person``` object.  The copy will be assigned to new variable ```printUserBound```.  The argument passed into the ```bind``` parenthesis is the name of the object that the ```this``` variable in the copied function will point to.
+
+
+```javascript
+var printUserBound = printUser.bind(person)
+```
+We now have new function ```printUserBound``` which is an exact copy of ```printUser```, **except** when the line ```console.log(this.getFullName())``` is run ```this``` refers to the ```person``` object.  So it runs ```console.log(person.getFullName())``` and will successfully log ```John Smith```.  Running ```printUser()``` will still return an error as this function remains unchanged.
+
+When invoking ```printUserBound``` we can also pass arguments into the parenthesis for the ```num``` argument.  For example, the following code would log both ```John Smith``` and the number ```4``` to the console.
+
+```javascript
+printUserBound(2)
+```
+
+### call()
+#### Syntax
+```aFunction.call(anObject, arg1, arg2, ...argX)```
+
+`aFunction` refers to a function that will be invoked and `anObject` is the object to reference with the invoked functions `this` variable.  If `aFunction` was configured to accept arguments, these can be passed **after** passing in `anObject`.
+</br>
+</br>
+
+Typing ```.call()``` on the end of a function simply invokes it, the same way that ```()``` would.  However, inside ```call```s parenthesis, we can specify what any instances of ```this``` in the code will refer to.
+
+```printUser.call(person)``` will invoke the ```printUser``` function with its ```this``` reference set to the ```person``` object.  Notice the function is run immediately.  It is not copied to a new variable as we saw with ```bind```.
+
+
+### apply()
+#### Syntax
+```aFunction.apply(anObject, [anArray])```
+
+`aFunction` refers to a function that will be invoked and `anObject` is the object to reference with the invoked functions `this` variable.  If `aFunction` was configured to accept arguments, these can be passed **after** passing in `anObject`.
+</br>
+</br>
+
+Typing ```.apply()``` on the end of a function work exactly the same way as ```.call()``` with one difference - The arguments must be passed in an array.
+
+```javascript
+printUser.apply(person, [arg1, arg2, ...argX])
+```
+___
+
+## Common Use Cases
+
+### Method Sharing
+The ```call``` and ```apply``` methods can be leveraged to shorten and simplify our code drastically by sharing repeatable methods between multiple objects.  In the example below, I have created three objects for storing people, all of which we are going to assume require a method to print the first and last name.
```

Comment:
take out leveraged to shorten and simplify - better to use 'can be used to simplify and shorten code significantly'